library/conn_bridge.c

```diff
@@ -124,15 +165,41 @@ extern int ziti_conn_bridge_fds(ziti_connection conn, uv_os_fd_t input, uv_os_fd
     br->data = br;
     br->fdbr = fdbr;
 
-    uv_handle_set_data((uv_handle_t *) br->input, br);
+    uv_handle_set_data(br->input, br);
     ziti_conn_set_data(conn, br);
 
     ziti_conn_set_data_cb(conn, on_ziti_data);
-    uv_read_start(br->input, bridge_alloc, on_input);
+    uv_read_start((uv_stream_t *) br->input, bridge_alloc, on_input);
 
     return ZITI_OK;
 }
 
+static void on_bridge_idle(uv_timer_t *t) {
+    struct ziti_bridge_s *br = t->data;
+    ZITI_LOG(DEBUG, "closing bridge conn[%d.%d] due to idle timeout", br->conn->ziti_ctx->id, br->conn->conn_id);
+    close_bridge(br);
+}
+
+int ziti_conn_bridge_idle_timeout(ziti_connection conn, unsigned long millis) {
+    struct ziti_bridge_s *br = ziti_conn_data(conn);
+    if (millis == 0) {
+        br->idle_timeout = 0;
+        if (br->idle_timer) {
+            uv_close((uv_handle_t *) br->idle_timer, (uv_close_cb) free);
+            br->idle_timer = NULL;
+        }
+    } else {
+        br->idle_timeout = millis;
```

Comment:
i feel like this should have a floor too. we shouldn't allow values lower than some threshold, or at very least log a warning when/if that happens. That'll be hard to track down if someone puts a value very low in there